kayak_core/src/context.rs

```diff
@@ -557,4 +560,27 @@ impl KayakContext {
     pub fn force_release_cursor(&mut self) -> Option<Index> {
         self.event_dispatcher.force_release_cursor()
     }
+
+    pub fn cursor_icon(&self) -> CursorIcon {
+        self.cursor_icon
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn set_cursor_icon(&mut self, icon: CursorIcon) {
+        self.cursor_icon = icon;
+    }
+
+    fn update_cursor(&mut self) {
+        if self.event_dispatcher.hovered.is_none() {
+            return;
+        }
+
+        let hovered = self.event_dispatcher.hovered.unwrap();
+        if let Some(node) = self.widget_manager.nodes.get(hovered) {
+            if let Some(node) = node {
+                let icon = node.resolved_styles.cursor.resolve();
+                self.cursor_icon = icon;
+            }
+        }
+    }
```

Comment:
Should this be moved outside of `KayakContext`? Does it make more sense in `EventDispatcher`? Or a new structure entirely?

Reply:
Let's leave it for now. I think maybe we can separate out some logic with a few traits maybe.